### PR TITLE
mass notify contest participants

### PR DESCRIPTION
**Pull request by ComputerGenius152**
closes https://github.com/jdabtieu/CTFOJ/issues/44

### CODE REVIEW

File: application.py [hunk 1]
```diff
@@ -513,6 +513,27 @@ def contest(contest_id):
...
+def contest_notify(contest_id):
+    if request.method == "GET":
+        return render_template('admin/contestnotify.html')
+
+    subject = request.form.get("subject")
+    if not subject:
+        return "Must provide subject", 400
+    message = request.form.get("message")
+    if not message:
+        return "Must provide message", 400
```
Comment (jdabtieu) [inline application.py:527]: replace `return "Must provide X", 400` with
```py3
flash('Must provide X', 'danger')
return render_template('admin/contestnotify.html'), 400
```

File: application.py [hunk 2]
```diff
@@ -513,6 +513,27 @@ def contest(contest_id):
...
+
+    data = db.execute("SELECT email FROM contest_users JOIN users on user_id=users.id WHERE contest_users.contest_id=:cid ORDER BY points DESC, lastAC ASC",
```
Comment (jdabtieu) [inline application.py:529]: don't think it's necessary to order by points and lastAC, consider getting rid of the ORDER BY part if it's not necessary.

**Review by jdabtieu — CHANGES_REQUESTED** [review]

**Comment by jdabtieu** [comment]
I also noticed that when sending the emails, all recipients also see everyone else who the email has been sent to, which is a privacy breach. It would be best to send emails individually if possible.

**Comment by ComputerGenius152** [comment]
Good catch, it now bcc's everyone.

**Review by jdabtieu — APPROVED** [review]

**Pull request merged**